src/XRP/signer.ts

```diff
@@ -44,8 +44,8 @@ const signer = {
    * @returns A set of bytes representing the inputs and a signature.
    */
   signTransaction(
-    transaction: Transaction,
-    wallet: Wallet,
+    transaction?: Transaction,
+    wallet?: Wallet,
   ): Uint8Array | undefined {
     if (transaction === undefined || wallet === undefined) {
```

Comment:
This `if` statement means we need to make the parameters above optional, otherwise it would be "impossible" for them to be `undefined` (because TS has an unsound type system).

Reply:
These params shouldn't be optional, not sure why I did this. Can we remove the if statement?